web/README.md

```diff
@@ -0,0 +1,418 @@
+
+### Conversion
+
+`/conversion` is a GET API which returns table wise rate of conversion which is encoded in color values.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+Table wise rate of conversion encoded in color values.
+
+Example
+
+```json
+{
+  "Albums": "GREEN",
+  "Singers": "BLUE",
+  "Userinfo": "RED"
+}
+```
+
+### Typemap
+
+(1) `/typemap` is a GET API which returns the source type to list of spanner type map, which can be used for manual type maping from the UI.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+Map from source type to list of possible spanner type and issue.
+
+Example
+
+```json
+{
+  "smallint": [
+    {
+      "T": "INT64",
+      "Brief": "Some columns will consume more storage in Spanner"
+    },
+    {
+      "T": "STRING",
+      "Brief": "Some columns will consume more storage in Spanner"
+    }
+  ],
+  "varchar": [
+    {
+      "T": "BYTES",
+      "Brief": ""
+    },
+    {
+      "T": "STRING",
+      "Brief": ""
+    }
+  ]
+}
+```
+
+(2) `/typemap/global` is a POST API which converts from source type map to desired spanner typemap globally (i.e, it affects all the tables). It takes a map from source type to spanner type in json format.
+
+#### Method
+
+`POST`
+
+#### Request body
+
+Map from source type to spanner type.
+
+Example
+
+```json
+{
+  "smallint": "STRING",
+  "varchar": "BYTES"
+}
+```
+
+#### Response body
+
+Updated Conv struct in JSON format.
+
+(3) `/typemap/table?table=<table_name>` is a POST API which performs following operations on a single table.
+
+- Remove column
+- Rename column
+- Remove or Add primary key
+- Update type of column
+- Remove or Add NOT NULL constraint
+
+#### Method
+
+`POST`
+
+#### Request body
+
+Column wise actions to be performed with following possible values for each field.
+
+- Removed : true/false
+- Rename : New name or empty string
+- PK : "" | "ADDED" | "REMOVED"
+- NotNull : "" | "ADDED" | "REMOVED"
+- ToType : New spanner type or empty string
+
+Example
+
+```json
+{
+  "UpdateCols": {
+    "AlbumTitle": {
+      "Removed": false,
+      "Rename": "AlbumName",
+      "PK": "",
+      "NotNull": "ADDED",
+      "ToType": "BYTES"
+    }
+  }
+}
+```
+
+#### Response body
+
+Updated Conv struct in JSON format.
+
+### Filepaths
+
+`/filepaths` is a GET API which generates and returns file paths for schema and report file.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+Paths to schema and report files.
+
+Example
+
+```json
+{
+  "Report": "/path/to/reportfile.report.txt",
+  "Schema": "/path/to/schemafile.schema.txt"
+}
+```
+
+### Interleave tables
+
+`/checkinterleave/table?table=<table_name>` is a GET API which checks it it is possible to convert this table into interleaved table in spanner, if it is possible it converts and returns parent table name, otherwise it returns failure message.
```

Comment:
Suggest rephrasing:

`/checkinterleave/table?table=<table_name>` is a GET API which checks whether it is
possible to convert a table into a Spanner interleaved table. If this conversion is possible,
then the schema is changed the parent table name is returned.
If the conversion is not possible, a failure message is returned.

Reply:
updated statement